.github/workflows/validate_pr.yml

```diff
@@ -0,0 +1,73 @@
+# Workflow that will reject non-contributors submitting PRs. The PRs will be auto-closed if the
+# submitter is not a contributor of the project.
+
+name: Validate PR Authorization
+on: [pull_request_target]
+jobs:
+  check_access:
+    runs-on: ubuntu-latest
+    name: Reject Unauthorized PRs
+
+    steps:
+      - name: Reject if submitter is not collaborator
```

Comment:
nit - contributor? (collaborator makes me think of WWII)

Reply:
Although it looks like that's how the API describes it, so maybe it's fine?

Reply:
Agree the terminology makes it sound a lot like we warped 80 years into the past, but per Github terminology:

* A collaborator is someone given increased access to a repo. Any collaborator that submits something is also a contributor.
* A contributor is anyone attempting to submit something to a repo, but they may not be a collaborator.

So in this case it's collaborators we're validating (contributors with explicit access rights).

Reply:
Yeah, makes sense.